resources/migrations/030_add_new_field_types.yaml

```diff
@@ -0,0 +1,130 @@
+                  name: is_fk
+                  type: boolean
+                  defaultValueBoolean: false
+                  constraints:
+                    nullable: false
+        - sql:
+            sql: UPDATE metabase_field SET is_fk = true WHERE special_type = 'fk'
+        - addColumn:
+            tableName: metabase_field
+            columns:
+              - column:
+                  name: is_category
+                  type: boolean
+                  defaultValueBoolean: false
+                  constraints:
+                    nullable: false
+        - addColumn:
+            tableName: metabase_field
+            columns:
+              - column:
+                  name: new_base_type
+                  type: varchar(256)
+                  defaultValue: 'type/*'
+                  constraints:
+                    nullable: false
+        - sql:
+            sql: UPDATE metabase_field SET is_category = true WHERE special_type = 'category'
+        - sql:
+            sql: UPDATE metabase_field SET new_base_type = 'type/collection.array' WHERE base_type ='ArrayField'
+        - sql:
+            sql: UPDATE metabase_field SET new_base_type = 'type/number.integer.big' WHERE base_type ='BigIntegerField'
+        - sql:
+            sql: UPDATE metabase_field SET new_base_type = 'type/boolean' WHERE base_type ='BooleanField'
+        - sql:
+            sql: UPDATE metabase_field SET new_base_type = 'type/text' WHERE base_type ='CharField'
+        - sql:
+            sql: UPDATE metabase_field SET new_base_type = 'type/datetime.date' WHERE base_type ='DateField'
+        - sql:
+            sql: UPDATE metabase_field SET new_base_type = 'type/datetime' WHERE base_type ='DateTimeField'
+        - sql:
+            sql: UPDATE metabase_field SET new_base_type = 'type/number.float.decimal' WHERE base_type ='DecimalField'
+        - sql:
+            sql: UPDATE metabase_field SET new_base_type = 'type/collection.map' WHERE base_type ='DictionaryField'
+        - sql:
+            sql: UPDATE metabase_field SET new_base_type = 'type/number.float' WHERE base_type ='FloatField'
+        - sql:
+            sql: UPDATE metabase_field SET new_base_type = 'type/number.integer' WHERE base_type ='IntegerField'
+        - sql:
+            sql: UPDATE metabase_field SET new_base_type = 'type/text' WHERE base_type ='TextField'
+        - sql:
+            sql: UPDATE metabase_field SET new_base_type = 'type/datetime.time' WHERE base_type ='TimeField'
+        - sql:
+            sql: UPDATE metabase_field SET new_base_type = 'type/text.uuid' WHERE base_type ='UUIDField'
+        - sql:
+            sql: UPDATE metabase_field SET new_base_type = 'type/*' WHERE base_type ='UnknownField'
+        - dropColumn:
+            tableName: metabase_field
+            columnName: base_type
+        - renameColumn:
+            tableName: metabase_field
+            oldColumnName: new_base_type
+            newColumnName: base_type
```

Comment:
renaming columns is something we should never need to do in migration scripts.  it's always easier to just create a new column, copy the necessary data over, then delete the old column whenever we want.

and i think we should avoid having any drop column statements until we get to major version changes.  the reason being that dropping a column creates a 100% guarantee that you can't revert from one version to a previous one.  Hopefully not something that users never need to do but it's good to keep it possible.